cellxgene_resources/curation_qa.ipynb

```diff
@@ -400,46 +400,41 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "def TENx_barcode_checker(df, num_to_check):\n",
-    "    if df.shape[0] < num_to_check:\n",
-    "        num_to_check = df.shape[0]\n",
-    "\n",
-    "    ref_dir = 'ref_files/'\n",
-    "    v2_file = '737K-august-2016.txt'\n",
-    "    v3_file = '3M-february-2018.txt'\n",
-    "    multiome_file = '737K-arc-v1.txt'\n",
-    "\n",
-    "    v2_list = [line.strip() for line in open(ref_dir + v2_file, 'r')]\n",
-    "    v3_list = [line.strip() for line in open(ref_dir + v3_file, 'r')]\n",
-    "    multiome_list = [line.strip() for line in open(ref_dir + multiome_file, 'r')]\n",
-    "\n",
-    "    cellcount = df.index.shape[0]\n",
-    "    barcode_pattern = '[ACTG]{16}'\n",
-    "    barcode_results = ''\n",
-    "    if re.search(barcode_pattern, df.index[0]):\n",
-    "        cellcount\n",
-    "        random_indices = [randint(0, cellcount - 1) for p in range(0, num_to_check)]\n",
-    "        barcodes = {'3pv2_5pv1_5pv2': 0,'3pv3': 0,'multiome': 0,'multiple': 0,'none': 0}\n",
-    "        for i in random_indices:\n",
-    "            if re.search(barcode_pattern, df.index[i]):\n",
-    "                barcode = re.search(barcode_pattern, df.index[i]).group(0)\n",
-    "                if barcode in v2_list:\n",
-    "                    if barcode in multiome_list:\n",
-    "                        barcodes['multiple'] += 1\n",
-    "                    elif barcode in v3_list:\n",
-    "                        barcodes['multiple'] += 1\n",
-    "                    else:\n",
-    "                        barcodes['3pv2_5pv1_5pv2'] += 1\n",
-    "                elif barcode in multiome_list:\n",
-    "                    if barcode in v3_list:\n",
-    "                        barcodes['multiple'] += 1\n",
-    "                    else:\n",
-    "                        barcodes['multiome'] += 1\n",
-    "                elif barcode in v3_list:\n",
-    "                    barcodes['3pv3'] += 1\n",
-    "                else:\n",
-    "                    barcodes['none'] += 1\n",
-    "        return barcodes"
+    "def summarize(v2,v3,m,b):\n",
+    "    if b == 1:\n",
+    "        if v2 + v3 + m > 1:\n",
+    "            return 'multiple'\n",
+    "        elif (v2 == 1) and (v3 == 0) and (m == 0):\n",
+    "            return '3pv2_5pv1_5pv2'\n",
+    "        elif (v2 == 0) and (v3 == 1) and (m == 0):\n",
+    "            return '3pv3'\n",
+    "        elif (v2 ==0 ) and (v3 == 0) and (m == 1):\n",
+    "            return 'multiome'\n",
+    "    else:\n",
+    "        pass"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def TENx_barcode_checker(obs_df, num_to_check):\n",
+    "    ref_df = pd.read_csv('../cellxgene_resources/10X_barcode_table.csv', sep=',', header=0, index_col='barcode')\n",
```

Comment:
the csv can go into cellxgene_resources/ref_files
and since the qa notebook can be assumed to be run from cellxgene_resources, it can be `ref_files/10X_barcode_table.csv`

Reply:
Actually, might even want to move defining ref_df to outside the function, so the notebook loads it only once, instead of every time the function is called. (it's big so it's not a super quick load and the time could add up for big datasets)
So unde `results = []` below probably makes sense